tests/test_api.py

```diff
@@ -231,7 +231,8 @@ def it_should_create_links(config_with_link):
             expect(os.listdir()).contains('my_link')
 
         def it_should_not_overwrite_files(config_with_link):
-            os.system("touch my_link")
+            os.system("mkdir my_link")
+            os.system("touch mylink/my_link")
```

Comment:
Since this changes the behavior of the test, I think this should be a new test:

- Keep `it_should_not_overwrite_files` as it was
- Add `it_should_not_overwrite_nested_files` with this logic